oresat_configs/__init__.py

```diff
@@ -1,19 +1,17 @@
 """OreSat OD database"""
 
+from __future__ import annotations
+
 # Checks that pyyaml is installed correctly. For performance reasons it must use the libyaml C
 # bindings. To use them both libyaml must be installed on the local system, and pyyaml must have
 # been built to use them. This works correctly on x86 systems, but on arm pyyaml is built by
 # default to not include the bindings.
-try:
-    from yaml import CLoader
-except ImportError as e:
-    raise ImportError(
-        "pyyaml missing/installed without libyaml bindings. See oresat-configs README.md for more"
-    ) from e
-
-from dataclasses import dataclass
+import yaml
+
+if not hasattr(yaml, "CLoader"):
+    raise ImportError("pyyaml installed without libyaml bindings. See oresat-configs README.md")
```

Comment:
Another way to check if `libyaml` is installed:

```suggestion
try:
    import _yaml
catch ModuleNotFoundError:
    log.error('libyaml is not installed on this system!')
```

Which is a meta package of `libyaml` and should produce an `ModuleNotFoundError` if not found on the system.

I know using try-catch to surround imports is not great but it might be more contextually correct in this situation given that this is a non-python dependency.

Reply:
You can see in the diff that the original check was of that style but the linters will then complain about unused imports. This check is being done here instead of in the modules that deal with yaml in order to fail fast, import time was significant in some C3 configurations and we'd like to know installations were correct sooner rather than later.

Additionally we specifically only care about `CLoader` so this is more direct.